qir/qat/ValidationPass/FunctionValidationPass.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Logging/ILogger.hpp"
+#include "Profile/Profile.hpp"
+#include "QatTypes/QatTypes.hpp"
+#include "StaticResourceComponent/StaticResourceComponentConfiguration.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <functional>
+#include <unordered_map>
+#include <vector>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    class FunctionValidationPass : public llvm::PassInfoMixin<FunctionValidationPass>
+    {
+      public:
+        using Instruction = llvm::Instruction;
+        using Value       = llvm::Value;
+        using ILoggerPtr  = ILogger::ILoggerPtr;
+        using Location    = ILogger::Location;
+        using StringRef   = llvm::StringRef;
+
+        enum ResourceType
+        {
+            None,
+            Qubit,
+            Result
+        };
```

Comment:
(nit) It is somewhat surprising to see an `enum` added, but not used in the _newly added_ code. I assume this enum is planned to be used in the future.

Reply:
It is a mistake that it is there.